File: lab-06.py
class Queue:
    def __init__(self):
        self.items = []

    def is_empty(self):
        return self.items == []

    def enqueue(self, item):
        self.items.append(item)

    def dequeue(self):
        if not self.is_empty():
            return self.items.pop(0)
        else:
            print("Queue is empty")
            return None

    def size(self):
        return len(self.items)

    def peek(self):
        if not self.is_empty():
            return self.items[0]
        else:
            print("Queue is empty")
            return None


# Example usage:
q = Queue()

# Enqueue items
q.enqueue(1)
q.enqueue(2)
q.enqueue(3)
q.enqueue(4)

# Peek at the front element
print("Front element:", q.peek())

# Dequeue items
print("Dequeued item:", q.dequeue())
print("Dequeued item:", q.dequeue())

# Size of the queue
print("Size of the queue:", q.size())

# Check if the queue is empty
print("Is queue empty?", q.is_empty())

File: Lab(ICE-2202)CPP/lab-06.cpp
#include <iostream>
#include <queue>
using namespace std;

class Queue
{
private:
    queue<int> q;

public:
    void enqueue(int value)
    {
        q.push(value);
    }

    void dequeue()
    {
        if (!q.empty())
        {
            q.pop();
        }
        else
        {
            cout << "Queue is empty." << endl;
        }
    }

    int front()
    {
        if (!q.empty())
        {
            return q.front();
        }
        else
        {
            cout << "Queue is empty." << endl;
            return -1;
        }
    }

    int size()
    {
        return q.size();
    }

    bool is_empty()
    {
        return q.empty();
    }
};

int main()
{
    Queue q;

    // Enqueue elements
    q.enqueue(1);
    q.enqueue(2);
    q.enqueue(3);

    // Print front element
    cout << "Front element: " << q.front() << endl;

    // Dequeue elements
    q.dequeue();
    q.dequeue();

    // Print size of the queue
    cout << "Size of the queue: " << q.size() << endl;

    // Check if queue is empty
    cout << "Is queue empty? " << (q.is_empty() ? "Yes" : "No") << endl;

    return 0;
}
